packages/fast-components-styles-msft/src/utilities/color/recipe.ts

```diff
@@ -0,0 +1,113 @@
+import {
+    checkDesignSystemResolver,
+    DesignSystem,
+    DesignSystemResolver,
+} from "../../design-system";
+import { backgroundColor } from "../design-system";
+import { Swatch, SwatchFamily } from "./common";
+import {
+    findClosestSwatchIndex,
+    findSwatchIndex,
+    getSwatch,
+    isDarkMode,
+    Palette,
+    swatchByContrast,
+} from "./palette";
+
+/**
+ * Resolves the index that the contrast search algorithm should start at
+ */
+function referenceColorInitialIndexResolver(
+    referenceColor: string,
+    sourcePalette: Palette,
+    designSystem: DesignSystem
+): number {
+    return findClosestSwatchIndex(sourcePalette, referenceColor)(designSystem);
+}
+
+function contrastTargetFactory(
+    targetContrast: number
+): (instanceContrast: number) => boolean {
+    return (instanceContrast: number): boolean => instanceContrast >= targetContrast;
+}
+
+function indexToSwatchFamily(
+    accessibleIndex: number,
+    palette: Palette,
+    direction: number,
+    restDelta: number,
+    hoverDelta: number,
+    activeDelta: number
+): SwatchFamily {
+    // One of the indexes will be rest, the other will be hover. Depends on the offsets and the direction.
+    const accessibleIndex2: number =
+        accessibleIndex + direction * Math.abs(restDelta - hoverDelta);
+
+    const indexOneIsRestState: boolean =
+        direction === 1
+            ? restDelta < hoverDelta
+            : direction * restDelta > direction * hoverDelta;
+
+    const restIndex: number = indexOneIsRestState ? accessibleIndex : accessibleIndex2;
+    const hoverIndex: number = indexOneIsRestState ? accessibleIndex2 : accessibleIndex;
+
+    const activeIndex: number = restIndex + direction * activeDelta;
+
+    return {
+        rest: getSwatch(restIndex, palette),
+        hover: getSwatch(hoverIndex, palette),
+        active: getSwatch(activeIndex, palette),
+    };
+}
+
+/**
+ * Function to derive accessible colors from contrast and delta configuration.
+ * Performs a simple contrast check against the colors and returns
+ * the color that has the most contrast against the background. If contrast
+ * cannot be retrieved correctly, function returns black.
+ */
+export function accessibleAlgorithm(
+    palette: Palette | DesignSystemResolver<Palette>,
+    minContrast: number | DesignSystemResolver<number>,
+    restDelta: number | DesignSystemResolver<number>,
```

Comment:
Same question for deltas - at some level it enables circumventing the DesignSystem which could be a bad thing.

Reply:
With this being an underlying abstraction I'm less concerned about circumventing the design system, but the reason it is here specifically is for cases where we needed a constant value or the value was already resolved, so avoiding doing it twice. For the constant, sometimes we want "0". I suppose we could write `() => 0`.

Reply:
Are there cases where this information *isn't* quantified in the DesignSystem?

Reply:
At some point more of the attributes might be in the design system, like the min-contrast, but for now they were hard-coded. For instance, foreground text is 14:1 and the rest offset is 0. We only have attributes for hover and active deltas.